File: frontend/src/types/Conversation.ts
import { ChatMessage } from './Message';

export interface ConversationPreview {
  id?: number;
  title: string;
  lastMessage?: string | null;
  lastMessageTime?: string | null;
  participants?: string;
}

export interface Conversation extends ConversationPreview {
  messages: ChatMessage[];
  createdAt?: string;
  updatedAt?: string;
}

export interface CreateConversationDto {
  title: string;
  participants: string;
  messages: ChatMessage[];
}

File: src/main/java/com/classy4j/chat/model/Conversation.java
package com.classy4j.chat.model;

import java.time.LocalDateTime;
import java.util.List;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import lombok.Data;

@Data
@Entity
public class Conversation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    private String title;
    private String lastMessage;
    private LocalDateTime lastMessageTime;
    private String participants;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    
    @OneToMany(mappedBy = "conversationId")
    private List<ChatMessage> messages;
} 